### DISCUSSION TITLE
How do you find when the daily and night rates start/finish

### DISCUSSION THREAD

**Discussion by swaapc** [BottlecapDave/HomeAssistant-OctopusEnergy#339] [General]
Hi,

I am on Flexible Octopus November 2022 v1 (tariff_code: E-2R-VAR-22-11-01-B, product_code: VAR-22-11-01).

I am having an impossible task getting Octopus to define when the start and finish of night/day time is for the above tariff. Their current answer is 00:30 to 07:30 all year round (ie the same in summer and winter). I note from your Home assistant Octopus Integration, that I am using, that the off peak starts at 00:30GMT and finishes at 07:30GMT (ie 01:30 to 8:30BST) which is what I would expect. However Octopus customer service seem adamant that it is from 00:30 to 07:30 (summer and winter)

I have had a look at your code and can't see how you find the peak and off peak times from Octopus's API. Maybe I am missing something. Can you give me a clue how you determine when the Peak and off peak times start and finish.

Regards
Chris Swaap

**Comment by BottlecapDave** [comment]
Hi there. There is some logic in the api client that determines when a day/night cycle starts/ends which can be found at https://github.com/BottlecapDave/HomeAssistant-OctopusEnergy/blob/develop/custom_components/octopus_energy/api_client.py#L998. This logic was taken from OE's support pages at https://octopus.energy/help-and-faqs/articles/what-happens-to-my-economy-seven-e7-tariff-when-i-have-a-smart-meter-installed/. Other tariffs helpfully provide the separations via the API.

**Comment by swaapc** [comment]
Thanks for that.

Interestingly I have tried over the last 'two months' to get a statement out of Octopus to what exactly the times are. This has so far proven impossible. I have had answers such as  0:30 to 4:30, 01:00 to 08:00 and 0:30 to 7:30. However I seem to word the question I never get a sensible answer back from them. For instance the latest response to my "is the 0:30 to 7:30 times GMT (UTC) or BST" is "With a smart meter, the times will update automatically to adjust between GMT and BST. This didn't used to be the case with traditional meters but you have a connected and working smart meter so it will adjust automatically." This may be the case but just doesn't sound right. I assume the 0:30 to 7:30 is GMT and that of course doesn't change.

I was hoping you might have found a method of extracting the start/finish of peak/offpeak times from the API.

**Reply by BottlecapDave** [reply]
Unfortunately not, the API for 2 rate meters is less than ideal

**Comment by swaapc** [comment]
Thanks for the info. I will continue my battle with Octopus.
